File: table-graph/src/__generated/services/TableGroupController.ts
import type { Executor } from '../';
import type { GenTableGroupCommonInput, GenTableGroupCommonView, GenTableGroupMoveInput, GenTableGroupTreeView, TableGroupQuery } from '../model/static';

export class TableGroupController {
    
    constructor(private executor: Executor) {}
    
    async create(options: TableGroupControllerOptions['create']): Promise<
        GenTableGroupCommonView
    > {
        let _uri = '/table/group/';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as GenTableGroupCommonView
    }
    
    async delete(options: TableGroupControllerOptions['delete']): Promise<number> {
        let _uri = '/table/group/';
        return (await this.executor({uri: _uri, method: 'DELETE', body: options.body})) as number
    }
    
    async edit(options: TableGroupControllerOptions['edit']): Promise<
        GenTableGroupCommonView
    > {
        let _uri = '/table/group/';
        return (await this.executor({uri: _uri, method: 'PUT', body: options.body})) as GenTableGroupCommonView
    }
    
    async getTrees(options: TableGroupControllerOptions['getTrees']): Promise<
        ReadonlyArray<GenTableGroupTreeView>
    > {
        let _uri = '/table/group/tree';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.groupIds?.join(',');
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'groupIds='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        return (await this.executor({uri: _uri, method: 'GET'})) as ReadonlyArray<GenTableGroupTreeView>
    }
    
    async move(options: TableGroupControllerOptions['move']): Promise<
        GenTableGroupCommonView
    > {
        let _uri = '/table/group/move';
        return (await this.executor({uri: _uri, method: 'PUT', body: options.body})) as GenTableGroupCommonView
    }
    
    async query(options: TableGroupControllerOptions['query']): Promise<
        ReadonlyArray<GenTableGroupCommonView>
    > {
        let _uri = '/table/group/query';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as ReadonlyArray<GenTableGroupCommonView>
    }
}

export type TableGroupControllerOptions = {
    'create': {readonly body: GenTableGroupCommonInput},
    'delete': {readonly body: ReadonlyArray<number>},
    'edit': {readonly body: GenTableGroupCommonInput},
    'getTrees': {readonly groupIds?: ReadonlyArray<number>},
    'move': {readonly body: GenTableGroupMoveInput},
    'query': {readonly body: TableGroupQuery}
}

File: src/main/kotlin/top/potmot/controller/TableGroupController.kt
package top.potmot.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import top.potmot.model.dto.GenTableGroupCommonInput
import top.potmot.model.dto.GenTableGroupCommonView
import top.potmot.model.dto.GenTableGroupMoveInput
import top.potmot.model.dto.GenTableGroupTreeView
import top.potmot.model.query.TableGroupQuery
import top.potmot.service.TableGroupService

@RestController
@RequestMapping("/table/group")
class TableGroupController (
    @Autowired val tableGroupService: TableGroupService
) {
    @PostMapping
    fun create(@RequestBody group: GenTableGroupCommonInput): GenTableGroupCommonView {
        return tableGroupService.createGroup(group)
    }

    @PutMapping
    fun edit(@RequestBody group: GenTableGroupCommonInput): GenTableGroupCommonView {
        return tableGroupService.editGroup(group)
    }

    @PutMapping("move")
    fun move(@RequestBody group: GenTableGroupMoveInput): GenTableGroupCommonView {
        return tableGroupService.moveGroup(group)
    }

    @GetMapping("/tree")
    fun getTrees(@RequestParam(required = false) groupIds: List<Long>?): List<GenTableGroupTreeView> {
        return tableGroupService.getTableTrees(groupIds)
    }

    @PostMapping("/query")
    fun query(@RequestBody query: TableGroupQuery): List<GenTableGroupCommonView> {
        return tableGroupService.queryGroups(query)
    }

    @DeleteMapping
    fun delete(@RequestBody ids: List<Long>): Int {
        return tableGroupService.deleteGroups(ids)
    }
}
